pkgs/servers/x11/xorg/generate-expr-from-tarballs.pl

```diff
@@ -73,8 +74,13 @@
         next;
     }
 
+    # split by first occurence of only numbers and dots surrounded by hyphens or end of line
+    my ($name, $version) = split(/(-[\.0-9]+(?:$|-(?:.+)))/, $pkgName, 2);
+
     $pkgURLs{$pkg} = $tarball;
-    $pkgNames{$pkg} = $pkgName;
+    $pkgNames{$pkg} = $name;
+    # remove minus from version
+    $pkgVersions{$pkg} = substr($version, 1);
```

Comment:
might be best to avoid that by starting the regex with `/-(`, especially given that capturing the entire match reads odd.

More generally, I think a lookahead might be a nicer way to implement this since it does not require the reader to know the behavior of `split` when captures are present. It also avoids the `.*`: `/-(?=[.0-9]+(?:$|-))`

Reply:
this is the exakt regex i was looking for. but hat trouble finding it. Thanks for your suggestion